pkg/cid/interfaces.go

```diff
@@ -36,6 +36,11 @@ type ClientIdentity interface {
 	// with a value of `attrValue`; otherwise, an error is returned.
 	AssertAttributeValue(attrName, attrValue string) error
 
+	// HasOUValue returns true if the client's subject DN contains an OU with the value
```

Comment:
Semantic versioning does not allow for interface changes. This incompatibly changes the contract as existing implementers no longer satisfy it. While we have not yet officially transitioned from v0 of the module, we have to be sensitive to the changes.

This is why idiomatic go prefers concrete types with interfaces defined at the consumer instead of the implementor...

Instead of extending the interface, we can create a new one that includes the method and we can perform a type assertion in the production code. This would provide the desired function without incompatible changes.

Reply:
Had an exchange with @braduf on rocket chat. Based on that conversation:

- clientIdentityImpl will be exported as `ClientID`
- New will return `*ClientID` instead of the `ClientIdentity` interface
- `ClientIdentity` will not be changed
- `HasOUValue` will be implemented at the package level and on `*ClientID`

While changing the return type from New is not ideal, it does not impact existing users or implementors of the interface.

We should also deprecate the interface at this point as it is unlikely we will be further extending it and there are no explicit references to it in other `fabric` or `fabric-chaincode-go` projects.

Reply:
@sykesm: Thanks for the clear list of changes. If I interpreted them well, they are now implemented.